utils/docker/Dockerfile.centos.7

```diff
@@ -12,13 +12,65 @@ LABEL maintainer="daos@daos.groups.io"
 # Intermittent cache-bust.  Used to reduce load on the actual CACHEBUST later.
 ARG CB0
 
+# If a local distro repository is supplied, then we should use it
+# and disable using the distro repositories.
+# The Docker image starts out with the DISTRO provided GPG keys
+# on the disk, but not installed.  These must be isntalled before
+# adding any repo that needs GPG keys but does not provide meta
+# data for installing those keys.  Then the epel-release
+# package must be installed and the GPG key it provides must be
+# installed.
+ARG REPO_URL=""
+ARG REPO_DISTRO=""
+RUN if [ -n "$REPO_DISTRO" ]; then                              \
+      rpm --import /etc/pki/rpm-gpg/* ;                         \
+      MY_REPO="${REPO_URL}${REPO_DISTRO}/";                     \
+      MY_NAME="${MY_REPO#*//}";                                 \
+      MY_NAME="${MY_NAME//\//_}";                               \
+      echo -e "[${MY_NAME}]\n\
+name=created from ${MY_REPO}\n\
+baseurl=${MY_REPO}\n\
+enabled=1\n\
+repo_gpgcheck=0\n\
+gpgcheck=1\n" >> /etc/yum.repos.d/local-centos-group.repo;      \
+      yum -y install --disablerepo=extras                       \
+             dnf dnf-plugins-core epel-release-7-11;             \
+      rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-7;         \
+      yum clean all;                                            \
+      dnf clean all;                                            \
```

Comment:
Why is this inside the `if [ -n "$REPO_DISTRO" ]`?  It seems like we want to do this regardless of whether we use a local repo or not.

Reply:
> This is inside the $REPO_DISTRO check so that the epel-release and dnf can be installed from $REPO_DISTRO instead of the upstream.

I'm not suggesting you do otherwise.  I am suggesting that installing DNF is orthogonal to adding a local repo and that it should be done afterward so that it applies to both the local and non-local repo cases equally -- minimizing the code in the `if` for the local repo to only doing what needs doing to adding the local repo and disabling the (duplicate) upstream repos.  Then the step that installs DNF is the same for both use-cases.

> I did not think it was worth the effort to eliminate the duplicate install of epel-release and dnf at the later step.

This is not the principle of least surprise IMHO.  Installing DNF is orthogonal to using a local repo or not and the result here is having a DNF installation at two unrelated areas of the process.

It should not be included in that step or installed at a later step if the local repo addition step is not done.  The way the CentOS 8 Dockerfile is structured to do the DNF install is much more clear, IMHO.  The way it's being done here is that future work on this file will have to know that DNF is being installed in two different places depending on the local repo addition, and IMHO, nobody is going to go looking for the second one after seeing the first one.

Reply:
Updated to have a else clause to be consistent with the CentOS 8 else clause.